bubble-wrap-style.yaml

```diff
@@ -2446,37 +2436,154 @@ layers:
                         - shield_text: true
                 draw:
                     mapzen_icon_library:
-                        #texture: mapzen_icon_library_shields_usa
+                        # texture: mapzen_icon_library_shields_usa
                         # you need to match any custom shield to the vector tile `network` values
-                        sprite: function() { return (feature.network + '-' + feature.shield_text.length + 'char'); }
-                        # FYI: sprite_default doesn't support functions, default is carried by parent style's sprite function
+                        sprite_default: |
+                            function() {
+                              switch (feature.shield_text.length) {
+                                case 1: return 'generic_shield-1char';
+                                case 2: return 'generic_shield-2char';
+                                case 3: return 'generic_shield-3char';
+                                case 4: return 'generic_shield-4char';
+                                case 5: return 'generic_shield-5char';
+                               }
+                            }
 
                 # US Interstate roads
                 US-I:
                     filter:
                         network: 'US:I'
-                        shield_text: true
                     draw:
                         mapzen_icon_library:
                             priority: 46
-                            # always show (or not show), irrespective of zoom
-                            visible: global.sdk_road_shields
+                            sprite: |
+                                function() {
+                                  switch (feature.shield_text.length) {
+                                    case 1: return 'US:I-1char';
```

Comment:
Hmm. That's may be fine to expand just for this USA example here with switch statements... but seems rather excessive to bloat the full USA and international shield theme imports. Those could easily get 5x as large as they are today in terms of line count. It's not impossible since much of that is auto-generated, but my goodness!

If the switch is more performant... could we say instead:

```
                                case 1: return feature.network + '-1char';
                                case 2: return feature.network + '-2char';
                                case 3: return feature.network + '-3char';
                                case 4: return feature.network + '-4char';
                                case 5: return feature.network + '-5char';
```

Reply:
This would still create new strings for each feature in js context which is quite expensive (at least it is in Duktape where all strings are interned, meaning the new string is hashed and then checked for identical strings to have only one instance in memory) 
What I thought of was that we could provide a stringbuilder function in js so that we could have a more efficient native implementation. This would require to return feature properties as proxy string objects instead of js strings, etc implementation details...

On the js side this could look like: ` return stringbuilder(feature.network, '-1char');